### PR TITLE
Optional LWJGLX debugging toggled via iris.properties + client commands

### PR DESCRIPTION
**Pull request by IMS212**
PRing this so we can delete the branch and just have it. It’s just good.

### CODE REVIEW

File: src/main/java/net/coderbot/iris/config/IrisConfig.java [hunk 1]
```diff
@@ -26,11 +26,17 @@
 	 */
 	private boolean enableShaders;
 
+	/**
+	 * If debug features should be enabled. Gives much more detailed GLSL error outputs at the cost of performance.
```
Comment (coderbot16) [inline src/main/java/net/coderbot/iris/config/IrisConfig.java:30]:
```suggestion
	 * If debug features should be enabled. Gives much more detailed OpenGL error outputs at the cost of performance.
```

File: src/main/java/net/coderbot/iris/Iris.java [hunk 2]
```diff
@@ -116,6 +117,11 @@ public static void onRenderSystemInit() {
 			return;
 		}
 
+		if (irisConfig.isDebugEnabled()) {
+			logger.warn("Debug features enabled, GLSL logging will be more verbose!");
```
Comment (coderbot16) [inline src/main/java/net/coderbot/iris/Iris.java:121]:
```suggestion
			logger.info("Debug features enabled, OpenGL error logging will be more verbose!");
```

File: src/main/java/net/coderbot/iris/gl/GLDebug.java [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+		StackTraceElement[] elems = throwable.getStackTrace();
+		StackTraceElement[] filtered = new StackTraceElement[elems.length];
+		int j = 0;
+		for (int i = offset; i < elems.length; i++) {
+			String className = elems[i].getClassName();
+			if (className == null) {
+				className = "";
+			}
+			if (className.startsWith("org.lwjglx.") && !className.startsWith("org.lwjglx.debug.opengl")
+					&& !className.startsWith("org.lwjglx.debug.glfw")) {
```
Comment (coderbot16) [inline src/main/java/net/coderbot/iris/gl/GLDebug.java:51]: This seems like something that needs to be updated.

**Comment by coderbot16** [comment]
This doesn't appear to actually compile.

**Comment by IMS212** [comment]
Fixed

**Comment by IMS212** [comment]
New client commands have been added via /iris, notably to toggle debugging, if shaders are enabled, and reloading without hotkeys.

**Comment by BeansBeefBroccoli** [comment]
Are the hotkeys going to stay?

**Review by coderbot16 — APPROVED** [review]
Looks good, thanks.

**Pull request merged**